src/quantum/tp6-32bit.js

```diff
@@ -0,0 +1,89 @@
+import _ from 'lodash';
+import React from 'react';
+import { withStyles } from '@material-ui/core';
+import PropTypes from 'prop-types';
+import { PAGES } from './config';
+import DashboardPage from '../components/DashboardPage/index';
+import PerfherderGraphContainer from '../containers/PerfherderGraphContainer/index';
+
+
+const styles = {
+  // This div helps with canvas size changes
+  // https://www.chartjs.org/docs/latest/general/responsive.html#important-note
+  body: {
+    backgroundColor: 'white',
+  },
+  area: {
+    display: 'flex',
+  },
+  column: {
+    flex: '50%',
+    alignItems: 'center',
+  },
+  chart: {
+    justifyContent: 'center',
+    padding: '1rem',
+
+  },
+
+};
+
+const TP6_32 = ({ classes }) => {
+  const allCharts = _
+    .chain(PAGES.data)
+    // ZIP HEADER WITH ROWS TO GET OBJECTS
+    .map(row => _.zipObject(PAGES.header, row))
+    // GROUP BY title
+    .groupBy(row => row.title)
+    // LOOP OVER EACH KEY/VALUE
+    .toPairs()
+    .map(([title, series], _i) => {
+      return (
+        <div className={classes.chart}>
+          <PerfherderGraphContainer
+          // key={`pages${i}`}
+            title={title}
+            series={
+            _
+              .chain(series)
+              .map((s) => {
+                return {
+                  label: s.label,
+                  seriesConfig: s,
+                };
+              })
+              .value()
+          }
+          />
+        </div>
+      );
+    })
+    // SPLIT INTO TWO
+    .chunk(2)
+    // TRANSPOSE, SO WE HAVE TWO COLUMNS
+    .unzip()
+    .value();
+
+  return (
+    <div className={classes.body}>
+      <DashboardPage title='TP6 - Page load on 32bit'>
+        <div className={classes.area}>
```

Comment:
Another way you can approach this is the following:

- Create a `PerfherderGraphs` component
- Pass the data you generate with `lodash` to it (e.g. <PerfherderGraphs graphs={yourData} />
- And you write this:
```javascript
const { yourData } = this.state;
return (
  <div className={classes.body}>
      <DashboardPage title='TP6 - Page load on 32bit'>
         <PerfherderGraphs graphs={yourData} />
      </DashboardPage>
  </div>
);
```

By doing this you will be separating the concern of handling the data and simply presenting it.

Reply:
I think this is solved by moving the list comprehension to the constructor